src/helics/core/helicsCLI11.hpp

```diff
@@ -153,7 +155,22 @@ class helicsCLI11App: public CLI::App {
             },
             "display system information details");
     }
-
+    void add_config_validation()
+    {
+        auto* opt = get_option("--config");
+        if (opt != nullptr) {
+            validate_positionals();
+            opt->check([](const std::string& fname) {
+                static const std::set<std::string> validExt = {
+                    ".ini", ".toml", ".json", ".INI", ".JSON", ".TOML"};
+                auto ext = std::filesystem::path(fname).extension().string();
```

Comment:
Should we just make the extension lowercase, so we only need to list one variant in the list of valid extensions?

Reply:
I thought about that but it seemed like it would be more efficient to just let the set deal with it, rather than call an additional function every time.  I tried routing it through the CLI validator that does that but it had some annoying side effects so I went away from that option.